x-pack/plugins/fleet/server/services/package_policy.ts

```diff
@@ -539,9 +538,32 @@ class PackagePolicyService implements PackagePolicyServiceInterface {
   ): Promise<DeletePackagePoliciesResponse> {
     const result: DeletePackagePoliciesResponse = [];
 
-    for (const id of ids) {
+    const packagePolicies = await this.getByIDs(soClient, ids, { ignoreMissing: true });
+    if (!packagePolicies) {
+      return [];
+    }
+
+    const uniqueAgentPolicyIds = [...new Set(packagePolicies.map((pP) => pP.policy_id))];
```

Comment:
nit: what do you of renaming `pP` => `packagePolicy` ? seems a little explicit to me